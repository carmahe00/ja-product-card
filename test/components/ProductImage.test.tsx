import React from 'react'
import renderer from 'react-test-renderer'
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';


describe('ProductImage', () => {
    

    test('debe de mostrar el componente correctamente con la imagen personalizado', () => {
        const wrapper = renderer.create(
            <ProductImage img="https://static1.mileroticos.com/photos/d/2022/02/15/c5/0d5145369dec455f03771f14fa19adac.jpg" className="custom-class" />
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('debe de mostrar el componente correctamente con imagen producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2} >
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})
