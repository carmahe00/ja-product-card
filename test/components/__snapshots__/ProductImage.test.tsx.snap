// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente correctamente con imagen producto 1`] = `
<div
  className="productCard undefined "
>
  <img
    alt="Producto"
    className="productImg undefined"
    src="./coffe-mug.png"
  />
</div>
`;

exports[`ProductImage debe de mostrar el componente correctamente con la imagen personalizado 1`] = `
<img
  alt="Producto"
  className="productImg custom-class"
  src="https://static1.mileroticos.com/photos/d/2022/02/15/c5/0d5145369dec455f03771f14fa19adac.jpg"
/>
`;
